source/Handlebars.Test/IssueTests.cs

```diff
@@ -220,5 +224,97 @@ public void SlashesInTemplateKey()
 
             Assert.Equal("hello world", result);
         }
+        
+        
+        // issue: https://github.com/Handlebars-Net/Handlebars.Net/issues/399
+        [Fact]
+        public void PassesWhenMemberAccessorIsNull()
+        {
+            var template = "{{Join ['a', \"b \",  42] ':'}}";
+
+            var handlebars = Handlebars.Create();
+            handlebars.RegisterHelper(new JoinHelper());
+
+            var handlebarsTemplate = handlebars.Compile(template);
+            var actual = handlebarsTemplate("");
+            
+            Assert.Equal("a:b :42", actual);
+        }
+
+        // issue: https://github.com/Handlebars-Net/Handlebars.Net/issues/402
+        [Fact]
+        public void NestedObjectIteration()
+        {
+            const string template = @"
+                {{#with test}}
+
+					{{#if complexItems}}
+					<ul>
+						{{#each complexItems}}
+							<li>{{name}} {{value}} {{evenMoreComplex.name}} {{evenMoreComplex.abbr}}</li>
+						{{/each}}
+					</ul>
+					{{/if}}
+
+				{{/with}}";
+            
+            var data = new
+            {
+                test = new
+                {
+                    complexItems = new[] {
+                        new { name = "a", value = 1, evenMoreComplex = new { name = "zzz", abbr = "z" } },
+                        new { name = "b", value = 2, evenMoreComplex = new { name = "yyy", abbr = "y" } },
+                        new { name = "c", value = 3, evenMoreComplex = new { name = "xxx", abbr = "x" } }
+                    },
+                }
+            };
+
+            var handlebars = Handlebars.Create();
+            var handlebarsTemplate = handlebars.Compile(template);
+
+            var result = handlebarsTemplate(data);
+            
+            const string expected = "<ul><li>a 1 zzz z</li><li>b 2 yyy y</li><li>c 3 xxx x</li></ul>";
+            var actual = result
+                .Replace("\n", string.Empty)
+                .Replace("\r", string.Empty)
+                .Replace("\t", string.Empty);
+            
+            Assert.Equal(expected, actual);
+        }
+        
+        private class JoinHelper : IHelperDescriptor<HelperOptions>
```

Comment:
@StefH 
Some inspiration on allocation-less implementation of Join helper based on your example in Handlebars-Net/Handlebars.Net/issues/399